### DISCUSSION TITLE
Some questions regarding automatic formatting of PHP code

### DISCUSSION THREAD

**Discussion by szymach** [apache/netbeans#4229] [Q&A] [answered]
Hello!

I have been using Netbeans for quite some time now and I never quite figured out how to achieve some things with automatic formatting. What I mean by that are settings in the `Tools -> Options -> Editor -> Formatting` combined with `alt + shift + f`.

For example, when I have a long method definition like this:

```php
class SomeClass 
{
    public function someFunction(string $argumentA, int $argumentB, array $argumentC): void
    {
    }
}
```
I would like it to be formatted like so:

```php
class SomeClass 
{
    public function someFunction(
        string $argumentA, 
        int $argumentB, 
        array $argumentC
    ): void {
    }
}
```
Unfortunately, the closest thing I got is something like this:

```php
class SomeClass 
{
    public function someFunction(string $argumentA, 
        int $argumentB, array $argumentC): void 
    {
    }
}
```
Same goes for array initializing, method calling etc., can someone please explain if this is achievable with Netbeans settings for the editor, or do I need to use a tool like PHPCSFixer?

Another thing is, I would like to know if there is a way to configure blank spaces to be added after methods' closing brace, but not after the last one? Example:

```php
class SomeClass 
{
    public function someFunctionA(): void
    {
    }

    public function someFunctionB(): void
    {
    }
    // is there a way to prevent this blank line from being added?
}
```

Thank you for all the help.

**Comment by KacerCZ** [comment, marked as answer]
This kind of formatting is not supported yet.
There is open issue related to PSR standard formatting https://github.com/apache/netbeans/issues/4685
